### PR TITLE
PLT-170 Add Marconi indexer which indexes nonce per each epoch in marconi-chain-index

### PR DESCRIPTION
**Pull request by koslambrou**
* In the EpochSDD worker inside `Marconi.ChainIndex.Indexers`, creating
  ExtLedgerState instead of just LedgerState, because we need
  HeaderState (part of ExtLedgerState) to get the nonce for each epoch.

* Renamed the `Marconi.ChainIndex.Indexers.EpochSDD` module to
  `Marconi.ChainIndex.Indexers.EpochState`.

* In `Marconi.ChainIndex.Indexers.EpochState`, we now store the nonce of
  each epoch in a separate table, along with rollback and resuming
  capabilities.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Formatting, PNG optimization, etc. are updated
    - [x] Important changes are reflected in changelog.d of the affected packages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [x] Reviewer requested

### CODE REVIEW

**Comment by eyeinsky** [comment]
Is there any info on whether this indexes nonces such that they match dbsync?

**Comment by koslambrou** [comment]
> Is there any info on whether this indexes nonces such that they match dbsync?

Not any proof, but the code _does_ come directly from db-sync. We can provide proof when working on the story which compares db-sync with marconi.

**Comment by eyeinsky** [comment]
> > Is there any info on whether this indexes nonces such that they match dbsync?
> 
> Not any proof, but the code _does_ come directly from db-sync. We can provide proof when working on the story which compares db-sync with marconi.

Maybe use [Koios](https://api.koios.rest/#overview) then which exposes a REST API to dbsync for mainnet?

**Comment by koslambrou** [comment]
@eyeinsky Ready for review. Confirmed that the result matches db-sync on pre-prod by looking at epochs 4,5,10 and 20 (random epochs).

**Review by eyeinsky — APPROVED** [review]

File: marconi-chain-index/src/Marconi/ChainIndex/Indexers/EpochState.hs [hunk 1]
```diff
@@ -13,7 +13,7 @@
 --
 -- This module will create the SQL tables:
 --
--- + table: epoch_spd
+-- + table: epoch_sdd
```
[the remark this answers is not in the document]
Comment (koslambrou) [reply marconi-chain-index/src/Marconi/ChainIndex/Indexers/EpochState.hs:16]: You're right. It got lost. Going to add it back.

File: marconi-sidechain/src/Marconi/Sidechain/Api/Query/Indexers/EpochSDD.hs [hunk 2]
```diff
@@ -1,52 +1,52 @@
-module Marconi.Sidechain.Api.Query.Indexers.EpochSPD
+module Marconi.Sidechain.Api.Query.Indexers.EpochSDD
```
[the remark this answers is not in the document]
Comment (koslambrou) [reply marconi-sidechain/src/Marconi/Sidechain/Api/Query/Indexers/EpochSDD.hs:1]: SDD = Stakepool Delegation Distribution

**Pull request merged**